tests/integration/test_interface.py

```diff
@@ -6,77 +6,96 @@
     generate_decennial_census,
     generate_social_security,
     generate_taxes_w2_and_1099,
     generate_women_infants_and_children,
 )
-from pseudopeople.schema_entities import COLUMNS, DATASETS
+from pseudopeople.schema_entities import COLUMNS, DATASETS, Dataset
 
 # TODO: Move into a metadata file and import metadata into prl
 DATA_COLUMNS = ["year", "event_date", "survey_date", "tax_year"]
 
 
 @pytest.mark.parametrize(
-    "data_dir_name, noising_function, use_sample_data",
+    "dataset, noising_function, use_sample_data",
     [
-        (DATASETS.census.name, generate_decennial_census, True),
-        (DATASETS.census.name, generate_decennial_census, False),
-        (DATASETS.acs.name, generate_american_community_survey, True),
-        (DATASETS.acs.name, generate_american_community_survey, False),
-        (DATASETS.cps.name, generate_current_population_survey, True),
-        (DATASETS.cps.name, generate_current_population_survey, False),
-        (DATASETS.ssa.name, generate_social_security, True),
-        (DATASETS.ssa.name, generate_social_security, False),
-        (DATASETS.tax_w2_1099.name, generate_taxes_w2_and_1099, True),
-        (DATASETS.tax_w2_1099.name, generate_taxes_w2_and_1099, False),
-        (DATASETS.wic.name, generate_women_infants_and_children, True),
-        (DATASETS.wic.name, generate_women_infants_and_children, False),
-        ("DATASETS.tax_1040.name", "todo", True),
-        ("DATASETS.tax_1040.name", "todo", False),
+        (DATASETS.census, generate_decennial_census, True),
+        (DATASETS.census, generate_decennial_census, False),
+        (DATASETS.acs, generate_american_community_survey, True),
+        (DATASETS.acs, generate_american_community_survey, False),
+        (DATASETS.cps, generate_current_population_survey, True),
+        (DATASETS.cps, generate_current_population_survey, False),
+        (DATASETS.ssa, generate_social_security, True),
+        (DATASETS.ssa, generate_social_security, False),
+        (DATASETS.tax_w2_1099, generate_taxes_w2_and_1099, True),
+        (DATASETS.tax_w2_1099, generate_taxes_w2_and_1099, False),
+        (DATASETS.wic, generate_women_infants_and_children, True),
+        (DATASETS.wic, generate_women_infants_and_children, False),
+        ("DATASETS.tax_1040", "todo", True),
+        ("DATASETS.tax_1040", "todo", False),
     ],
 )
 def test_generate_dataset(
-    data_dir_name: str, noising_function: Callable, use_sample_data: bool, tmpdir
+    dataset: Dataset, noising_function: Callable, use_sample_data: bool, tmpdir
 ):
     """Tests that noised datasets are generated and as expected. The 'use_sample_data'
     parameter determines whether to use the sample data (if True) or
     a non-default root directory with multiple datasets to compile (if False)
     """
     if noising_function == "todo":
-        pytest.skip(reason=f"TODO: implement dataset {data_dir_name}")
+        pytest.skip(reason=f"TODO: implement dataset {dataset}")
```

Comment:
dataset is a Dataset, not a string.

Reply:
Not sure I understand.

Reply:
Oh I see, will change to `{dataset.name}`